pkg/apis/core/validation/validation.go

```diff
@@ -4396,12 +4399,25 @@ func validateSysctls(sysctls []core.Sysctl, fldPath *field.Path) field.ErrorList
 	return allErrs
 }
 
-// ValidatePodSecurityContext test that the specified PodSecurityContext has valid data.
-func ValidatePodSecurityContext(securityContext *core.PodSecurityContext, spec *core.PodSpec, specPath, fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+// validatePodSecurityContext verifies the SecurityContext of a Pod, but only
+// when it is a Pod and not an embedded PodSpec.
+func validatePodSecurityContext(securityContext *core.PodSecurityContext, spec *core.PodSpec, specPath, scPath *field.Path, opts PodValidationOptions) field.ErrorList {
```

Comment:
This name is pretty misleading, since this doesn't actually verify the whole security context, only the hostnetwork port bits, and relies on validatePodSpecSecurityContext being called separately. With a single caller, I'd probably just inline this into `validatePodMetadataAndSpec`:

```
	if securityContext != nil {
		allErrs = append(allErrs, validateHostNetwork(securityContext.HostNetwork, spec.Containers, specPath.Child("containers"))...)
	}
```

edit: after reviewing above, we still need to do *some* validation on hostnetwork ports in workload specs, so we may not need the separate function _or_ the inline if we plumb host port validation options down to a shared validation method. See comment above.

Reply:
I was trying to avoid retooling all of this , but ACK that it is lacking.  Will take a look at another commit